harness/determined/common/experimental/determined.py

```diff
@@ -61,9 +61,139 @@ def __init__(
+            return user.User(
+                user_id=resp.user.id,
+                username=resp.user.username,
+                admin=resp.user.admin,
+                session=self._session,
+                active=resp.user.active,
+                display_name=resp.user.displayName,
+                agent_uid=resp.user.agentUserGroup.agentUid,
+                agent_gid=resp.user.agentUserGroup.agentGid,
+                agent_user=resp.user.agentUserGroup.agentUser,
+                agent_group=resp.user.agentUserGroup.agentGroup,
+            )
+        else:
+            return user.User(
+                user_id=resp.user.id,
+                username=resp.user.username,
+                admin=resp.user.admin,
+                session=self._session,
+                active=resp.user.active,
+                display_name=resp.user.displayName,
+            )
+
+    def get_user_by_name(self, user_name: str) -> user.User:
+        resp = bindings.get_GetUserByUsername(session=self._session, username=user_name)
+        if resp.user.agentUserGroup is not None:
+            return user.User(
+                user_id=resp.user.id,
+                username=resp.user.username,
+                admin=resp.user.admin,
+                session=self._session,
+                active=resp.user.active,
+                display_name=resp.user.displayName,
+                agent_uid=resp.user.agentUserGroup.agentUid,
+                agent_gid=resp.user.agentUserGroup.agentGid,
+                agent_user=resp.user.agentUserGroup.agentUser,
+                agent_group=resp.user.agentUserGroup.agentGroup,
+            )
+        else:
+            return user.User(
+                user_id=resp.user.id,
+                username=resp.user.username,
+                admin=resp.user.admin,
+                session=self._session,
+                active=resp.user.active,
+                display_name=resp.user.displayName,
+            )
+
+    def whoami(self) -> user.User:
+        resp = bindings.get_GetMe(self._session)
+        if resp.user.agentUserGroup is not None:
+            return user.User(
+                user_id=resp.user.id,
+                username=resp.user.username,
+                admin=resp.user.admin,
+                session=self._session,
+                active=resp.user.active,
+                display_name=resp.user.displayName,
+                agent_uid=resp.user.agentUserGroup.agentUid,
+                agent_gid=resp.user.agentUserGroup.agentGid,
+                agent_user=resp.user.agentUserGroup.agentUser,
+                agent_group=resp.user.agentUserGroup.agentGroup,
+            )
+        else:
+            return user.User(
+                user_id=resp.user.id,
+                username=resp.user.username,
+                admin=resp.user.admin,
+                session=self._session,
+                active=resp.user.active,
+                display_name=resp.user.displayName,
+            )
```

Comment:
this block is repeated a few times. can we have it as a classmethod on `user.User` instead e.g.
```
@classmethod
def _from_bindings(raw: v1User):
    ...
```

Reply:
I wrote a method for this but I can't add it as a classmethod because self._session is a part of the instance.